examples/canvas/app/boot.js

```diff
@@ -0,0 +1,96 @@
+import Microcosm, { update } from 'microcosm'
+import prepareCanvas from './prepare-canvas'
+
+let size = 20
+let limit = Infinity
+let rows = Math.min(window.innerWidth / size | 0, limit)
+let columns = Math.min(window.innerHeight / size | 0, limit)
+let writes = size * 25
+
+let repo = new Microcosm()
+let advance = (x, y) => ({ x, y })
+
+repo.addDomain('pixels', {
+  getInitialState() {
+    return {}
+  },
+  rotateHue (n) {
+    return isNaN(n) ? 0 : n + 5
+  },
+  advance (state, { x, y }) {
+    return update(state, [x, y], this.rotateHue)
+  },
+  register() {
+    return {
+      [advance]: this.advance
+    }
+  }
+})
+
+var canvas = document.createElement('canvas')
+var context = canvas.getContext('2d')
+
+document.body.appendChild(canvas)
+
+prepareCanvas(canvas)
+
+context.scale(size, size)
+
+for (let x = 0; x < rows; x++) {
+  for (let y = 0; y < columns; y++) {
+    repo.on(`change:pixels.${x}.${y}`, hue => {
+      context.fillStyle = `hsl(${hue}, 70%, 60%)`
+      context.fillRect(x, y, 1, 1)
+    })
+  }
+}
+
+canvas.addEventListener('mousemove', (event) => {
+  let x = event.clientX / size | 0
+  let y = event.clientY / size | 0
+
+  repo.push(advance, x, y)
+})
+
+function randomMoves (n = 1) {
+  while (n-- > 0) {
+    let x = Math.random() * rows | 0
+    let y = Math.random() * columns | 0
+
+    repo.push(advance, x, y)
+  }
+}
+
+function updateLabel () {
+  let events = rows * columns * writes
```

Comment:
Ah event is a bad name. This is, generally, how many keys the system can watch per 16 ms. 

This actually probably needs to go, or some other measure needs to be put here. I added this back when the data binding traversing was linear.

So this is probably more like this can crawl "2,982" cells 494 times per frame.